Source/Scene/AutoExposure.js

```diff
@@ -0,0 +1,416 @@
+        var height = autoexposure._height;
+
+        var pixelFormat = PixelFormat.RGBA;
+        var pixelDatatype = context.halfFloatingPointTexture ? PixelDatatype.HALF_FLOAT : PixelDatatype.FLOAT;
+        var sampler = new Sampler({
+            wrapS : TextureWrap.CLAMP_TO_EDGE,
+            wrapT : TextureWrap.CLAMP_TO_EDGE,
+            minificationFilter : TextureMinificationFilter.NEAREST,
+            magnificationFilter : TextureMagnificationFilter.NEAREST
+        });
+
+        var length = Math.ceil(Math.log(Math.max(width, height)) / Math.log(3.0));
+        var framebuffers = new Array(length);
+        for (var i = 0; i < length; ++i) {
+            width = Math.max(Math.ceil(width / 3.0), 1.0);
+            height = Math.max(Math.ceil(height / 3.0), 1.0);
+            framebuffers[i] = new Framebuffer({
+                context : context,
+                colorTextures : [new Texture({
+                    context : context,
+                    width : width,
+                    height : height,
+                    pixelFormat : pixelFormat,
+                    pixelDatatype : pixelDatatype,
+                    sampler : sampler
+                })]
+            });
+        }
+
+        var lastTexture = framebuffers[length - 1].getColorTexture(0);
+        autoexposure._previousLuminance = new Framebuffer({
+            context : context,
+            colorTextures : [new Texture({
+                context : context,
+                width : lastTexture.width,
+                height : lastTexture.height,
+                pixelFormat : pixelFormat,
+                pixelDatatype : pixelDatatype,
+                sampler : sampler
+            })]
+        });
+
+        autoexposure._framebuffers = framebuffers;
+    }
+
+    function destroyCommands(autoexposure) {
+        var commands = autoexposure._commands;
+        if (!defined(commands)) {
+            return;
+        }
+
+        var length = commands.length;
+        for (var i = 0; i < length; ++i) {
+            commands[i].shaderProgram.destroy();
+        }
+        autoexposure._commands = undefined;
+    }
+
+    function createUniformMap(autoexposure, index) {
+        var uniforms;
+        if (index === 0) {
+            uniforms = {
+                colorTexture : function() {
+                    return autoexposure._colorTexture;
+                },
+                colorTextureDimensions : function() {
+                    return autoexposure._colorTexture.dimensions;
+                }
+            };
+        } else {
+            var texture = autoexposure._framebuffers[index - 1].getColorTexture(0);
+            uniforms = {
+                colorTexture : function() {
+                    return texture;
+                },
+                colorTextureDimensions : function() {
+                    return texture.dimensions;
+                }
+            };
+        }
+
+        uniforms.minimumLuminance = function() {
```

Comment:
Can these three uniforms be combined or is that hard to do in the post-process framework?